muxtools/subtitle/sub.py

```diff
@@ -118,42 +118,69 @@ def clean_garbage(self: SubFileSelf) -> SubFileSelf:
         self.__update_doc(doc)
         return self
 
-    def autoswapper(self: SubFileSelf, allowed_styles: list[str] | None = DEFAULT_DIALOGUE_STYLES, print_swaps: bool = False) -> SubFileSelf:
+    def autoswapper(
+        self: SubFileSelf,
+        allowed_styles: list[str] | None = DEFAULT_DIALOGUE_STYLES,
+        print_swaps: bool = False,
+        inline_marker: str = "*",
+        line_marker: str = "***",
+    ) -> SubFileSelf:
         """
         autoswapper does the swapping.
         Too lazy to explain
 
         :param allowed_styles:      List of allowed styles to do the swapping on
                                     Will run on every line if passed `None`
         :param print_swaps:         Prints the swaps
+        :param inline_marker:       Marker to use for inline swaps.
+                                    Should be one character. Default `*`
+        :param line_marker:         Marker to use for full-line swaps. Default `***`
 
         :return:                    This SubTrack
         """
         doc = self._read_doc()
 
         events = []
 
+        if type(inline_marker) != str or len(inline_marker) == 0:
+            warn("Given invalid inline marker. Using default '*'.", self)
+            inline_marker = "*"
+        elif len(inline_marker) > 1:
+            warn(f"Inline marker '{inline_marker}' should be one character. Using '{inline_marker[0]}'.", self)
```

Comment:
is there any reason for this condition? I wouldn't even have thought of making it customizable but might as well leave it entirely up to the user

Reply:
(and, yeah, if we implement full string escaping for the regex, then there's no reason _not_ to allow multi-char strings)